llvm/test/CodeGen/ARM/branch-folder-single-bb-crash.mir

```diff
@@ -0,0 +1,80 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
+# RUN: llc -mtriple=armv7-apple-ios --run-pass=branch-folder %s -o - | FileCheck %s
+
+## Don't crash in branch-folder's HoistCommonCodeInSuccs when TBB and FBB are
+## the same block.
```

Comment:
I think this comment needs to be detached from the implementation details and made broader -- "We hoist code from successor blocks, and in abnormal circumstances we can have control flow that branches to the same block -- test that we handle this correctly". Or something.